### PR TITLE
feat(base-config): add no-unknown-wire-adapters and no-unexpected-wire-adapter-usages

### PR DESCRIPTION
**Pull request by jodarove**
This PR adds and configures [@lwc/lwc/no-unknown-wire-adapters](https://github.com/salesforce/eslint-plugin-lwc/blob/master/docs/rules/no-unknown-wire-adapters.md) and [@lwc/lwc/no-unexpected-wire-adapter-usages](https://github.com/salesforce/eslint-plugin-lwc/blob/master/docs/rules/no-unexpected-wire-adapter-usages.md) rules to the base config.

### CODE REVIEW

File: base.js [hunk 1]
```diff
@@ -6,6 +6,113 @@
...
+        module: '@salesforce/**',
+        identifier: '*',
+    },
+];
+
+const WIRE_ADAPTERS_WITH_RESTRICTED_USE = [
+    {
+        module: 'lightning/communityNavigationMenuApi',
+        identifier: 'getCommunityNavigationMenu',
+    },
```
Comment (kevinv11n) [inline base.js:25]: This is not exposed on Lightning Platform
```suggestion
```

File: base.js [hunk 2]
```diff
@@ -6,6 +6,113 @@
...
+        identifier: 'MessageContext',
+    },
+    {
+        module: 'lightning/navigation',
+        identifier: 'CurrentPageReference',
+    },
+    {
+        module: 'lightning/uiActionsApi',
+        identifier: 'getLookupActions',
+    },
```
Comment (kevinv11n) [inline base.js:37]: This is not exposed on Lightning Platform
```suggestion
```

File: base.js [hunk 3]
```diff
@@ -6,6 +6,113 @@
...
+    {
+        module: 'lightning/uiLayoutApi',
+        identifier: 'getLayout',
+    },
+    {
+        module: 'lightning/uiLayoutApi',
+        identifier: 'getLayoutUserState',
+    },
```
Comment (kevinv11n) [inline base.js:45]: This is not exposed on Lightning Platform
```suggestion
```

File: base.js [hunk 4]
```diff
@@ -6,6 +6,113 @@
...
+    {
+        module: 'lightning/uiListApi',
+        identifier: 'getListUi',
+    },
+    {
+        module: 'lightning/uiLookupsApi',
+        identifier: 'getLookupRecords',
+    },
```
Comment (kevinv11n) [inline base.js:53]: This is not exposed on Lightning Platform
```suggestion
```

File: base.js [hunk 5]
```diff
@@ -6,6 +6,113 @@
...
+        identifier: 'getRecordActions',
+    },
+    {
+        module: 'lightning/uiRecordActionsApi',
+        identifier: 'getRecordEditActions',
+    },
+    {
+        module: 'lightning/uiRecordActionsApi',
+        identifier: 'getRelatedListRecordActions',
+    },
```
Comment (kevinv11n) [inline base.js:77]: This is not exposed on Lightning Platform
```suggestion
```

File: base.js [hunk 6]
```diff
@@ -6,6 +6,113 @@
...
+        identifier: 'getRelatedListRecords',
+    },
+    {
+        module: 'lightning/uiRelatedListApi',
+        identifier: 'getRelatedListsCount',
+    },
+    {
+        module: 'lightning/uiRelatedListApi',
+        identifier: 'getRelatedListsInfo',
+    },
```
Comment (kevinv11n) [inline base.js:113]: This is not exposed on Lightning Platform
```suggestion
```

**Comment by jodarove** [comment]
@kevinv11n your suggestion were accurate 🥇, the remaining list is those adapters exposed to customers.

**Pull request merged**